src/moscot/base/problems/problem.py

```diff
@@ -90,9 +90,8 @@ def _split_mass(arr: ArrayLike) -> ArrayLike:
             if subset is None:  # allow for numeric values
                 data = np.asarray(adata.obs[data], dtype=float)
             else:
-                if isinstance(subset, str):
-                    subset = [subset]
-                data = np.asarray(adata.obs[data].isin(subset), dtype=float)
+                sset = subset if isinstance(subset, list) else [subset]  # type:ignore[list-item]
```

Comment:
I still slightly prefer the `isinstance(..., str)` check, as it's less constraining than the `list` check.